python/lsst/sims/utils/CompoundCoordinateTransformations.py

```diff
@@ -0,0 +1,493 @@
+    of the exposure (rotTelPos is calculated at the beginning of the exposure;
+    expmjd is reckoned at the middle of the exposure).
+    """
+    altRad, azRad, paRad = _altAzPaFromRaDec(raRad, decRad, obs)
+
+    #20 March 2015
+    #I do not know where this expression comes from; we should validate it against
+    #the definitions of rotTelPos and rotSkyPos
+    return (rotSkyRad + paRad)%(2.*np.pi)
+
+
+def calcObsDefaults(raRad, decRad, altRad, azRad, rotTelRad, mjd, band, longRad, latRad):
+    """
+    Fromats input data into a dict of metadata that PhoSim expect.
+
+    Users should probably not be calling this by hand, as it makes no effort to ensure that
+    the input values are self-consistent.  This method is called by
+    makeObsParamsAzAltTel
+    makeObsParamsAzAltSky
+    makeObsPeramsRaDecTel
+    makeObsParamsRaDecSky
+    which do ensure self-consistency of values.
+
+    @param [in] raRad is RA in radians
+    (In the International Celestial Reference System)
+
+    @param [in] decRad is Dec in radians
+    (In the International Celestial Reference System)
+
+    @param [in] altRad is altitude in radians
+
+    @param [in] azRad is azimuth in radians
+
+    @param [in] rotTelRad is rotTelPos in radians
+
+    @param [in] mjd is the Universal Time expressed as an MJD
+
+    @param [in] band is 'u', 'g', 'r', 'i', 'z', or 'y'
+    (i.e. the bandpass of the observation)
+
+    @param [in] longRad is the observer's longitude in radians
+    (positive east of the prime meridan)
+
+    @param [in] latRad is the observer's latitude in radians
+    (positive north of the equator)
+
+    @param [out] a dict of meta data which PhoSim expects in the
+    headers of its input InstanceCatalogs
+    """
+    obsMd = {}
+    #Defaults
+    obsTemp = ObservationMetaData(mjd=mjd, site=Site(longitude=longRad, latitude=latRad))
+    moonra, moondec = _raDecFromAltAz(-np.pi/2., 0., obsTemp)
+    sunalt = -np.pi/2.
+    moonalt = -np.pi/2.
+    dist2moon = haversine(moonra, moondec, raRad, decRad)
+    obsMd['Opsim_moonra'] = moonra
+    obsMd['Opsim_moondec'] = moondec
+    obsMd['Opsim_sunalt'] = sunalt
+    obsMd['Opsim_moonalt'] = moonalt
+    obsMd['Opsim_dist2moon'] = dist2moon
+
+    rotSkyPos = _getRotSkyPos(raRad, decRad, obsTemp, rotTelRad)
+    obsMd['Opsim_filter'] = band
+    obsMd['pointingRA'] = raRad
+    obsMd['pointingDec'] = decRad
+    obsMd['Opsim_rotskypos'] = rotSkyPos
+    obsMd['Opsim_rottelpos'] = rotTelRad
+    obsMd['Unrefracted_Altitude'] = altRad
+    obsMd['Unrefracted_Azimuth'] = azRad
+    return obsMd
+
+
+def makeObservationMetadata(metaData):
+    return OrderedDict([(k,(metaData[k], np.asarray(metaData[k]).dtype))
+                         for k in metaData])
+
+
+def makeObsParamsAzAltTel(azRad, altRad, mjd, band, rotTelRad=0., longRad=-1.2320792, latRad=-0.517781017, **kwargs):
+    '''
+    Calculate a minimal set of observing parameters give the ra, dec, and time of the observation.
```

Comment:
Shouldn't this and the first line in the docstring for the next method say "given the alt, az, and time of the observation." rather than ra, dec, time?